core/src/main/java/org/apache/iceberg/io/OutputFileFactory.java

```diff
@@ -27,7 +28,7 @@
 import org.apache.iceberg.encryption.EncryptedOutputFile;
 import org.apache.iceberg.encryption.EncryptionManager;
 
-public class OutputFileFactory {
+public class OutputFileFactory implements Serializable {
```

Comment:
Because the `IcebergStreamWriter` will need the `RowDataTaskWriterFactory` to create `TaskWriter`,  it could pass it to `TaskWriterFactory` when constructing, that means all members inside `RowDataTaskWriterFactory` should be serializable (include itself), unfortunately `OutputFileFactory` is one of its member.

Reply:
`outputFileFactory` should be `transient` in RowDataTaskWriterFactory, it is inited in `initialize`.

Reply:
OK, it make sense.